src/Model/VerifyEmailSignatureComponents.php

```diff
@@ -25,10 +27,21 @@ final class VerifyEmailSignatureComponents
      */
     private $uri;
 
-    public function __construct(\DateTimeInterface $expiresAt, string $uri)
+    /**
+     * @var int timestamp when the signature was created
+     */
+    private $generatedAt;
+
+    /**
+     * @var int expiresAt translator interval
+     */
+    private $transInterval = 0;
+
+    public function __construct(\DateTimeInterface $expiresAt, string $uri, int $generatedAt = 0)
```

Comment:
And... one more time... is it "legal" in this bundle (even if it's a bad idea) to have a permanent expiration (meaning, a link that does NOT expire ever)?

Reply:
> Oh, and should $generatedAt be instead a \DateTimeInterface like $expiresAt?

It could be, but unlike `$expiresAt`, we're providing a getter for the `$generatedAt`. End of the day it could go either way, but requiring a timestamp vs a DateTimeInterface elevates any concerns of mutation I think. Less room for error.

>  is it "legal" in this bundle (even if it's a bad idea) to have a permanent expiration

technically no - but one could set the `lifetime` to 9999999999 or some giant number and achieve the same results..  setting `lifetime` to `0` would render the signature useless as it will have already expired before the logic even finishes through the compiler.

all of the above applies in reset as well.